prow.sh

```diff
@@ -942,16 +949,20 @@ install_e2e () {
         return
     fi
 
-    git_checkout "${CSI_PROW_E2E_REPO}" "${GOPATH}/src/${CSI_PROW_E2E_IMPORT_PATH}" "${CSI_PROW_E2E_VERSION}" --depth=1 &&
-    if [ "${CSI_PROW_E2E_IMPORT_PATH}" = "k8s.io/kubernetes" ]; then
-        patch_kubernetes "${GOPATH}/src/${CSI_PROW_E2E_IMPORT_PATH}" "${CSI_PROW_WORK}" &&
-        go_version="${CSI_PROW_GO_VERSION_E2E:-$(go_version_for_kubernetes "${GOPATH}/src/${CSI_PROW_E2E_IMPORT_PATH}" "${CSI_PROW_E2E_VERSION}")}" &&
-        run_with_go "$go_version" make WHAT=test/e2e/e2e.test "-C${GOPATH}/src/${CSI_PROW_E2E_IMPORT_PATH}" &&
-        ln -s "${GOPATH}/src/${CSI_PROW_E2E_IMPORT_PATH}/_output/bin/e2e.test" "${CSI_PROW_WORK}" &&
-        run_with_go "$go_version" make WHAT=vendor/github.com/onsi/ginkgo/ginkgo "-C${GOPATH}/src/${CSI_PROW_E2E_IMPORT_PATH}" &&
-        ln -s "${GOPATH}/src/${CSI_PROW_E2E_IMPORT_PATH}/_output/bin/ginkgo" "${CSI_PROW_BIN}"
+    if [ "${CSI_PROW_SIDECAR_E2E_IMPORT_PATH}" != "none" ]; then
```

Comment:
This can be replaced with `sidecar_tests_enabled`, right?

Reply:
I'm unclear about one thing regarding this PR: is the goal to run either a local E2E suite or the one from Kubernetes, or is the goal to support both in the same job?

Because of this if check here it seems to be the former, because `install_e2e` can only be used to install one suite, but not both. I think it would be better to support both suites in the same job.

Reply:
@pohly That makes sense. I've updated the code in the latest commit to run the tests in a single job. My original testing only ran the local tests, so I didn't realise that `install_e2e` is run only one (although now the `if` at the start of that function makes more sense :))

Reply:
@pohly tests running in the sidecar repos would potentially want to enable some feature gates in the container. I think this file would need further changes to support that. If it runs in a single job, the feature gates would be enabled for the Kubernetes tests as well.

Reply:
Unless of course we undo the change after the local tests are complete

Reply:
Does this apply to out of tree feature gates as well? We don't need to rebuild, just patch the particular sidecar manifest with the right arguments. 
Sidecar repos with alpha features would also contain e2e tests for those features. So certain feature gates in that sidecar only need to be enabled when running those tests, right?

Reply:
The feature gates for the driver are different, but isn't the driver also only getting deployed once for both E2E suites?

Reply:
True. But as with cluster feature gates I prefer simplicity over flexibility here unless we know that it is needed.

Reply:
So will we need a change to this `if` to enable out-of-tree features in the sidecar manifest?

Reply:
Okay, I'll need to take a deeper look into where to update the feature flags. Potentially here - https://github.com/kubernetes-csi/csi-release-tools/blob/master/prow.sh#L695 but let me play around with it a bit

Reply:
Should we move ahead with this PR in the meanwhile? I don't think there's an obvious solution to adding features gates in a particular sidecar container.

Reply:
We can do that. Can you squash commits and then try out that commit in some sidecar repo where a local E2E suite is defined? In other words, create a test PR in some other repo?

It will fail the unit check for locally modified `release-tools`, but that's fine.

Reply:
Created a PR here - https://github.com/kubernetes-csi/external-provisioner/pull/798
The local test is just a skeleton to prove that the tests within that repo will run with these changes.